pkg/tests/tasks/traffic/circuit_breaking_test.go

```diff
@@ -20,95 +20,110 @@ import (
-			log.Log.Errorf("Error response: %v", msg)
-		}
-
-		log.Log.Info("Tripping the circuit breaker")
+	NewTest(t).Id("T6").Groups(Full, InterOp).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+		ns := "bookinfo"
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, ns)
+		})
+
+		app.InstallAndWaitReady(t, app.Httpbin(ns), app.Fortio(ns))
+
+		t.Log("verify traffic management tripping circuit breaker")
+		t.LogStep("Configure circuit breaker destination rule")
+		oc.ApplyString(t, ns, httpbinCircuitBreaker)
+
+		t.LogStep("Verify connection with curl: expected 200 OK")
+		retry.UntilSuccess(t, func(t test.TestHelper) {
+			oc.Exec(t,
+				pod.MatchingSelector("app=fortio", ns),
+				"fortio",
+				"/usr/bin/fortio curl -quiet http://httpbin:8000/get",
+				assert.OutputContains("200",
+					"Got expected response from httpbin: 200 OK",
+					"ERROR: Got unexpected response from httpbin not 200 OK"))
+		})
+
+		t.LogStep("Tripping the circuit breaker")
 		connection := 2
 		reqCount := 50
-		tolerance := 0.5
-
-		command = fmt.Sprintf(`/usr/bin/fortio load -c %d -qps 0 -n %d -loglevel Warning http://httpbin:8000/get`, connection, reqCount)
-		msg, err = util.PodExec("bookinfo", pod, "fortio", command, false)
-		util.Inspect(err, "Failed to get response", "", t)
-
-		re := regexp.MustCompile(`Code 200.*`)
-		line := re.FindStringSubmatch(msg)[0]
-		re = regexp.MustCompile(`: [\d]+`)
-		word := re.FindStringSubmatch(line)[0]
-		c200, err := strconv.Atoi(strings.TrimLeft(word, ": "))
-		util.Inspect(err, "Failed to parse code 200 count", "", t)
-
-		re = regexp.MustCompile(`Code 503.*`)
-		line = re.FindStringSubmatch(msg)[0]
-		re = regexp.MustCompile(`: [\d]+`)
-		word = re.FindStringSubmatch(line)[0]
-		c503, err := strconv.Atoi(strings.TrimLeft(word, ": "))
-		util.Inspect(err, "Failed to parse code 503 count", "", t)
-
-		if util.IsWithinPercentage(c200, reqCount, 0.6, tolerance) && util.IsWithinPercentage(c503, reqCount, 0.4, tolerance) {
-			log.Log.Infof(
-				"Success. Circuit breaking acts as expected. "+
-					"Code 200 hit %d of %d, Code 503 hit %d of %d", c200, reqCount, c503, reqCount)
-		} else {
-			t.Errorf(
-				"Failed Circuit breaking. "+
-					"Code 200 hit %d 0f %d, Code 503 hit %d of %d", c200, reqCount, c503, reqCount)
-		}
-
-		log.Log.Info("Query the istio-proxy stats")
-		command = fmt.Sprintf(`pilot-agent request GET stats | grep httpbin | grep pending`)
-		msg, _ = util.PodExec("bookinfo", pod, "istio-proxy", command, false)
-		log.Log.Infof("%s", msg)
+		tolerance := 0.1
+		retry.UntilSuccess(t, func(t test.TestHelper) {
+			t.Log("To tipping the circuit breaker we are going to send 50 requests to httpbin with 2 connections")
+			msg := oc.Exec(t,
+				pod.MatchingSelector("app=fortio", ns),
+				"fortio",
+				fmt.Sprintf("/usr/bin/fortio load -c %d -qps 0 -n %d -loglevel Warning http://httpbin:8000/get", connection, reqCount))
+
+			t.LogStep("Validate the number of 200 responses")
```

Comment:
Why is this a step? We're only getting the number of responses here. We're not actually validating them.